rpcs3/Loader/TAR.cpp

```diff
@@ -217,11 +264,95 @@ bool tar_object::extract(std::string vfs_mp)
 	return true;
 }
 
-bool extract_tar(const std::string& file_path, const std::string& dir_path)
+std::vector<u8> tar_object::save_directory(const std::string& src_dir, std::string full_path, std::vector<u8>&& init)
+{
+	const std::string& target_path = full_path.empty() ? src_dir : full_path;
+
+	fs::stat_t stat{};
+	if (!fs::stat(target_path, stat))
+	{
+		return std::move(init);
+	}
+
+	u32 count = 0;
+
+	if (stat.is_directory)
+	{
+		u32 count = 0;
+
+		for (auto& entry : fs::dir(target_path))
+		{
+			if (entry.name.find_first_not_of('.') == umax) continue;
+
+			init = save_directory(src_dir, target_path + '/' + entry.name, std::move(init));
+			count++;
+		}
+
+		if (count)
+		{
+			return std::move(init);
+		}
+	}
+
+	auto write_octal = [](char* ptr, u64 i)
+	{
+		if (!i)
+		{
+			*ptr = '0';
+			return;
+		}
+
+		ptr += utils::aligned_div(std::bit_width(i), 3) - 1;
+
+		for (; i; ptr--, i /= 8)
+		{
+			*ptr = static_cast<char>('0' + (i % 8));
+		}
+	};
+
+	TARHeader header{};
+	std::memcpy(header.magic, "ustar3", 6);
+
+	const std::string_view saved_path{target_path.data() + src_dir.size(), target_path.size() - src_dir.size()};
+
+	// Prefer saving to name feild as much as we can
```

Comment:
Maybe it's better to follow existing standards for header extensions, in order to keep the tarball readable by existing tools.

Reply:
```suggestion
	// Prefer saving to name field as much as we can
```